### PR TITLE
feat(github-actions): create a github action to automatically apply and remove breaking changes label

### PR DESCRIPTION
**Pull request by josephperrott**
See individual commits.

### CODE REVIEW

File: github-actions/utils.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {getInput} from '@actions/core';
+const {getToken} = require('github-app-installation-token');
+
+export async function getAuthTokenForAngularRobotApp(): Promise<string>;
+export async function getAuthTokenForAngularRobotApp(
```
Comment (devversion) [inline github-actions/utils.ts:5]: Nit: Could you add a description? I specifically would be interested in how this token is preferred, compared to the token that is already available in Github actions? i.e. do we want to act as the ng robot, compared to the `github-actions` bot account?

File: github-actions/breaking-changes-label/main.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      issue_number: number,
+      name: breakingChangesLabel,
+    });
+    console.log(`Removed ${breakingChangesLabel} label from PR #${number}`);
+  }
+}
+
+// Only run if the action is executed in a repository with is in the Angular org. This is in place
+// to prevent the action from actually running in a fork of a repository with this action set up.
+if (context.repo.owner === 'josephperrott') {
```
Comment (devversion) [inline github-actions/breaking-changes-label/main.ts:49]:
```suggestion
if (context.repo.owner === 'angular') {
```

File: github-actions/breaking-changes-label/main.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  const hasBreakingChangeLabel = await (
+    await client.issues.listLabelsOnIssue({issue_number: number, owner, repo})
+  ).data.find((label) => label.name === breakingChangesLabel);
+  console.log('hasBreakingChangeLabel', hasBreakingChangeLabel);
+
+  const hasBreakingChangeCommit = (
+    await client.paginate(client.pulls.listCommits, {owner, pull_number: number, repo})
+  ).some((commit) => {
+    return parseCommitMessage(commit.commit.message).breakingChanges.length > 0;
```
Comment (devversion) [inline github-actions/breaking-changes-label/main.ts:24]: Nice! I love that we are able to use this helper from `ng-dev`!

**Review by devversion — APPROVED** [review]
LGTM

File: github-actions/breaking-changes-label/main.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import * as core from '@actions/core';
+import {context} from '@actions/github';
+import {Octokit} from '@octokit/rest';
+import {parseCommitMessage} from '../../ng-dev/commit-message/parse';
+import {getAuthTokenForAngularRobotApp} from '../utils';
+
+const breakingChangesLabel = 'breaking changes';
```
Comment (alan-agius4) [inline github-actions/breaking-changes-label/main.ts:7]: Actually it appears that the label in FW has been changed to `flag: breaking change`.

https://github.com/angular/angular/commit/ebc80b3b5c6d05d11093aadeab4c893119ee16cf#diff-0f6460592cf1db69dc570c128942fb78af50c792b862af847654785406727242

**Pull request closed without merging**